pkg/gofr/container/datasources.go

```diff
@@ -275,6 +275,38 @@ type MongoProvider interface {
 	provider
 }
 
+// SurrealDB defines an interface representing a SurrealDB client with common database operations.
+type SurrealDB interface {
+
+	// Query executes a Surreal query with the provided variables and returns the query results as a slice of interfaces.
+	// It returns an error if the query execution fails.
+	Query(ctx context.Context, query string, vars map[string]interface{}) ([]interface{}, error)
```

Comment:
Please use `any` everywhere


```suggestion
	Query(ctx context.Context, query string, vars map[string]any) ([]any, error)
```

This was added in Go 1.18 and improves the readability